syntax = "proto3";
package proto;

import "FileInfo.proto";
import "OpenFile.proto";
import "defs.proto";
// import "Histogram.proto";
import "ImageData.proto";
import "SpectralProfile.proto";
// import "SpatialProfile.proto";
// import "Region.proto";

service FitsReaders {
  rpc CheckStatus(Empty) returns  (StatusResponse) {}
  rpc OpenFile(OpenFileRequest) returns (StatusResponse) {}
  rpc CloseFile(FileCloseRequest) returns (StatusResponse) {}
  rpc GetFileInfo (FileInfoRequest) returns (FileInfoResponse) {}
  rpc GetImageDataStream (ImageDataRequest) returns (stream ImageDataResponse) {}
  rpc GetSpectralProfile (SpectralProfileReaderRequest) returns (SpectralProfileResponse) {}
}
// service FitsServices{
//   rpc CheckStatus(Empty) returns  (StatusResponse) {}
//   rpc OpenFile(OpenFileRequest) returns (OpenFileACK) {}
//   rpc CloseFile(FileCloseRequest) returns (StatusResponse) {}
//   rpc GetFileInfo (FileInfoRequest) returns (FileInfoResponse) {}
//   rpc CreateRegion (SetRegion) returns (SetRegionAck) {}
//   rpc GetImageDataStream (ImageDataRequest) returns (stream ImageDataResponse) {}
//   rpc GetSpatialProfile (SetSpatialReq) returns (SpatialProfileData) {}
//   rpc GetSpectralProfile (SpectralProfileRequest) returns (SpectralProfileResponse) {} 
//   rpc GetHistogram (SetHistogramReq) returns (HistogramResponse) {}
// }
// message Empty {}

// message StatusResponse {
//   bool status = 1;
//   string statusMessage = 2;
// }

// message FileOpenRequest {
//   string fileName = 1;
//   string hdu = 2;
//   string uuid = 3;
// }

// message FileCloseRequest {
//   string uuid = 2;
// }

// message FileInfoRequest {
//   string uuid = 1;
// }

// message FileInfoResponse {
//   int32 hduNum = 2;
//   int32 hduType = 3;
//   int32 dataType = 4;
//   repeated int32 hduShape = 5;
// }

// message ImageDataRequest {
//   string uuid = 1;
//   repeated int32 start = 2;
//   int64 numPixels = 3;
// }

// message ImageDataResponse {
//   bytes data = 1;
// }

// message SpectralProfileRequest {
//   string uuid = 1;
//   int32 x = 2;
//   int32 y = 3;
//   int32 z = 4;
//   int32 numPixels = 5;
//   int32 width = 6;
//   int32 height = 7;
// }

// message SpectralProfileResponse {
//   bytes data = 1;
// }